### PR TITLE
RHMAP-5721 - Update fh-fhc to use common ESLint rules

### PR DESCRIPTION
**Pull request by camilamacedo86**
JIRA: https://issues.jboss.org/browse/RHMAP-5721

### CODE REVIEW

**Comment by camilamacedo86** [comment]
please retest it

File: lib/cmd/fhc/target.js [hunk 1]
```diff
@@ -106,7 +106,13 @@ function target(argv, cb) {
...
       var _gotVersion = function() {
         ini.save(function(err) {
-          target.message = i18n._("Successfully targeted: ") + targ.target + (targ.user ? (" User: " + targ.user) : (" API Key: " + targ[keys.KEY_ID]));
+          var message;
+          if (targ.user) {
+            message = " User: " + targ.user;
+          } else {
+            message = " API Key: " + targ[keys.KEY_ID];
+          }
+          target.message = i18n._("Successfully targeted: " + targ.target + message);
```
Comment (grdryn) [inline lib/cmd/fhc/target.js]: Is this functionally the same as it was before, or different?

From my reading of the old code, only `"Successfully targeted: "` was in the string passed to `i18n._()`, but now there's more in the new code. Is this okay?

File: lib/utils/error-handler.js [hunk 2]
```diff
@@ -106,12 +106,21 @@ function writeLogFile(cb) {
...
-      , pref = m.pref ? " " + m.pref : ""
-      , b = lvl + pref + " "
-      , msg = typeof m.msg === "string" ? m.msg
-        : msg instanceof Error ? msg.stack || msg.message
-        : sys.inspect(m.msg, 0, 4);
+    var msg;
+    if (typeof m.msg === "string") {
+      msg =  m.msg;
+    } else {
+      if (msg instanceof Error) {
```
Comment (grdryn) [inline lib/utils/error-handler.js]: I think a level of nesting can be removed here by bringing this `if` up to the `else` above it to make an `else if`. Does that sound right?

**Comment by camilamacedo86** [comment]
@grdryn first tks for the review/comments.
I will check your comments. Before I made one PR as you are suggesting for each rule. However, if you look the comment in the JIRA @wtrocki told that all should be in the same PR.

File: lib/utils/error-handler.js (hunk 2 above)
Comment (camilamacedo86) [inline lib/utils/error-handler.js]: Done into the review for the next Pr.

File: lib/cmd/fhc/target.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/cmd/fhc/target.js]: If you broke step by step the ternary: -          

`target.message = i18n._("Successfully targeted: ") + targ.target + (targ.user ? (" User: " + targ.user) : (" API Key: " + targ[keys.KEY_ID]));`

It is exactly the result. Just change the ternary for if.

File: lib/cmd/fh2/app/act.js [hunk 3]
```diff
@@ -35,7 +35,14 @@ function act(argv, cb) {
     var app = res.app;
     var inst = res.inst;
     var widgId = app.guid;
-    var nodejs = 'boolean' === typeof inst.nodejs ? inst.nodejs : (inst.nodejs === 'true' ? true : false);
+
+    var nodejs;
+    if ('boolean' === typeof inst.nodejs) {
+      nodejs = inst.nodejs;
+    } else {
+      nodejs= inst.nodejs === 'true' ? true : false;
```
Comment (camilamacedo86) [inline lib/cmd/fh2/app/act.js]: Done for the next PR ( review )

File: lib/cmd/fhc/target.js (hunk 1 above)
Comment (grdryn) [inline lib/cmd/fhc/target.js]: I don't think it's exactly the same. To be exactly the same, the last line would instead be:

```javascript
target.message = i18n._("Successfully targeted: ") + targ.target + message;
```

I don't know enough about any of this to know which one is more correct. Maybe someone else who knows what's going on here could take a look. I'm confident that the rest of the PR is fine.

@pb82 @wei-lee maybe one of you would be in a better position than I to comment on this?

**Comment by grdryn** [comment]
> Before I made one PR as you are suggesting for each rule. However, if you look the comment in the JIRA @wtrocki told that all should be in the same PR.

It's hard to find the best balance to keep us all happy :)

What I'm suggesting is separate commits in one PR, rather than separate PRs. Like I said though, don't even consider going back and changing it now for this case. :)

File: lib/cmd/fhc/target.js (hunk 1 above)
Comment (wei-lee) [inline lib/cmd/fhc/target.js]: I think @grdryn  is right here. The `User :` or `API Key: ` part is not in the `i18n._` before the change. Better keep it the same.

**Comment by grdryn** [comment]
> Just a small suggestion if similar changes are made to other components in the future: keep the auto-gemerated changes in one commit, and create another commit for manual changes. This will make the review a lot easier.

This is what I was trying to say. 👍 For the manual changes, even a separate commit per manually-fixed rule would be nice. :)

File: lib/cmd/fhc/target.js (hunk 1 above)
Comment (camilamacedo86) [inline lib/cmd/fhc/target.js]: done:-) tks guys

**Pull request merged**